### PR TITLE
Testsuite - store login information via cookies

### PR DESCRIPTION
**Pull request by lkotek**
## What does this PR change?

PR documents possibility to use cookies to store login information instead of logging again using login form. It doesn't modify any functionality of testsuite. 

Purpose of PR is to save this solution for possible future use.

## Links

https://github.com/SUSE/spacewalk/issues/7655

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

**Comment by lkotek** [comment]
@Bischoff @srbarrios: Thanks for review to both ;-) I will address those changes after RC1 validation.

**Comment by lkotek** [comment]
Moved to testsuite docs.

File: testsuite/documentation/cucumber-steps.md [hunk 1]
```diff
@@ -737,3 +737,34 @@ When implementing a step, to convert a step host name into a target, use:
...
+  # This code is executed only in case session is not stored yet.
+  cookie = page.driver.browser.manage.all_cookies
+  # Serialization and save of cookies to file in binary mode
+  File.open(cookie_path, 'wb') { |cookie_file|
+    Marshal.dump(cookie, cookie_file)
+  }
+```
+Useful links:
+
+- [Selenium docs](https://www.seleniumhq.org/docs/03_webdriver.jsp)
```
Comment (Bischoff) [inline testsuite/documentation/cucumber-steps.md]: Maybe add a note that this is currently not being used, and why.

Comment (lkotek) [reply testsuite/documentation/cucumber-steps.md]: You're right! It is maybe the most important part about it :-) But it is only in the original issue.

**Comment by Bischoff** [comment]
> LGTM.
> But if we plan to use it we must have a card to address it, if we don't plan to use it, I don't see the need to have it in the documentation. We can add it when we need it.

We discussed it yesterday, and the conclusion was:

We don't plan to use it in the immediate, but we need it documented somewhere, so the knowledge is not lost if it resurrects.

This PR addresses this need.

**Comment by srbarrios** [comment]
> > LGTM.
> > But if we plan to use it we must have a card to address it, if we don't plan to use it, I don't see the need to have it in the documentation. We can add it when we need it.
> 
> We discussed it yesterday, and the conclusion was:
> 
> We don't plan to use it in the immediate, but we need it documented somewhere, so the knowledge is not lost if it resurrects.
> 
> This PR addresses this need.

We discussed it yesterday and the conclusion was to create a card to address it in the future.
@ktsamis can confirm.

**Comment by Bischoff** [comment]
> We discussed it yesterday and the conclusion was to create a card to address it in the future.
> @ktsamis can confirm.

To do what? There's no benefit in the immediate. There would only be benefit as far as we know if we get the test suite completely restructured, and that's not for tomorrow. I can remember we were saying we would do that *at that time*.

**Comment by Bischoff** [comment]
> We discussed it yesterday and the conclusion was to create a card to address it in the future.
> @ktsamis can confirm.

@ktsamis just told me in private he understood the same as me.

**Comment by ktsamis** [comment]
> > > LGTM.
> > > But if we plan to use it we must have a card to address it, if we don't plan to use it, I don't see the need to have it in the documentation. We can add it when we need it.
> > 
> > 
> > We discussed it yesterday, and the conclusion was:
> > We don't plan to use it in the immediate, but we need it documented somewhere, so the knowledge is not lost if it resurrects.
> > This PR addresses this need.
> 
> We discussed it yesterday and the conclusion was to create a card to address it in the future.
> @ktsamis can confirm.

Yes I thought we would open this card when we have the teststuite split into steps first, and when we see if we want to have sign in and sing out or we want cookies, then we open the card and we work on it. But I also thought we would open it at the time, not yet. This PR is fine for preserving the knowledge for that time if we want to enable the cookies.

**Comment by srbarrios** [comment]
Few months ago we did a refactor in that method to avoid the issue of been logged with another user, caused by cookies.
The best approach we could give at that moment was to check first if the user is logged with a different user and in that case log out, to then log in with the proper user. It makes the step slower adding [defensive code](http://www.erlang.se/doc/programming_rules.shtml#HDR11) when we shouldn't, I think that goes against our goal of idempotency features.
Now that we learnt how to manage cookies, my suggestion is to create a card to address it properly.

**Comment by Bischoff** [comment]
> Few months ago we did a refactor in that method to avoid the issue of been logged with another user, caused by cookies.
> The best approach we could give at that moment was to check first if the user is logged with a different user and in that case log out, to then log in with the proper user. It makes the step slower adding [defensive code](http://www.erlang.se/doc/programming_rules.shtml#HDR11) when we shouldn't, I think that goes against our goal of idempotency features.

That's what we said, yes.

> Now that we learnt how to manage cookies, my suggestion is to create a card to address it properly.

And we said we can do that in due time.

**Comment by Bischoff** [comment]
> The best approach we could give at that moment was to check first if the user is logged with a different user and in that case log out, to then log in with the proper user. It makes the step slower adding [defensive code](http://www.erlang.se/doc/programming_rules.shtml#HDR11) when we shouldn't, I think that goes against our goal of idempotency features.

On the technical side: considerations about defensive code to be avoided and idempotency are correct, but it's coding style considerations, not anything that would change performance, which was the original concern, and not anything urgent. Right now we're still at GMC.

We do keep similar things on hold, despite we burn of doing them, like reducing the core features, or making client and minion optional.

It will all get better after GMC. We will be able to restart refactorings and coding style issues.

**Pull request merged**